### PR TITLE
[SPARK-11102] [SQL] Uninformative exception when specifing non-exist …

### PR DESCRIPTION
**Pull request by zjffdu**
…input for JSON data source

Here's the exception after this patch

```
scala> val people = sqlContext.read.json("/Users/hadoop/github/spark/examples/src/main/resources/people.json")
15/10/16 07:41:10 INFO JSONRelation: Listing hdfs://0.0.0.0:9000/Users/jzhang/github/spark/examples/src/main/resources/users.parquet on driver
java.lang.RuntimeException: Input paths do not exist or are empty directories, Input Paths=hdfs://0.0.0.0:9000/Users/hadoop/github/spark/examples/src/main/resources/people.json
    at org.apache.spark.sql.execution.datasources.json.JSONRelation.org$apache$spark$sql$execution$datasources$json$JSONRelation$$createBaseRdd(JSONRelation.scala:91)
    at org.apache.spark.sql.execution.datasources.json.JSONRelation$$anonfun$6$$anonfun$apply$1.apply(JSONRelation.scala:109)
    at org.apache.spark.sql.execution.datasources.json.JSONRelation$$anonfun$6$$anonfun$apply$1.apply(JSONRelation.scala:109)
    at scala.Option.getOrElse(Option.scala:120)
    at org.apache.spark.sql.execution.datasources.json.JSONRelation$$anonfun$6.apply(JSONRelation.scala:109)
    at org.apache.spark.sql.execution.datasources.json.JSONRelation$$anonfun$6.apply(JSONRelation.scala:103)
    at scala.Option.getOrElse(Option.scala:120)
    at org.apache.spark.sql.execution.datasources.json.JSONRelation.dataSchema$lzycompute(JSONRelation.scala:103)
    at org.apache.spark.sql.execution.datasources.json.JSONRelation.dataSchema(JSONRelation.scala:102)
    at org.apache.spark.sql.sources.HadoopFsRelation.schema$lzycompute(interfaces.scala:561)
    at org.apache.spark.sql.sources.HadoopFsRelation.schema(interfaces.scala:560)
    at org.apache.spark.sql.execution.datasources.LogicalRelation.<init>(LogicalRelation.scala:37)
    at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:123)
    at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:107)
    at org.apache.spark.sql.DataFrameReader.json(DataFrameReader.scala:222)
    at $iwC$$iwC$$iwC$$iwC$$iwC$$iwC$$iwC$$iwC.<init>(<console>:19)
    at $iwC$$iwC$$iwC$$iwC$$iwC$$iwC$$iwC.<init>(<console>:24)
    at $iwC$$iwC$$iwC$$iwC$$iwC$$iwC.<init>(<console>:26)
    at $iwC$$iwC$$iwC$$iwC$$iwC.<init>(<console>:28)
    at $iwC$$iwC$$iwC$$iwC.<init>(<console>:30)
    at $iwC$$iwC$$iwC.<init>(<console>:32)
    at $iwC$$iwC.<init>(<console>:34)
    at $iwC.<init>(<console>:36)
    at <init>(<console>:38)
    at .<init>(<console>:42)
    at .<clinit>(<console>)
    at .<init>(<console>:7)
    at .<clinit>(<console>)
```

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala [hunk 1]
```diff
@@ -83,10 +83,13 @@ private[sql] class JSONRelation(
     val job = new Job(sqlContext.sparkContext.hadoopConfiguration)
     val conf = SparkHadoopUtil.get.getConfigurationFromJobContext(job)
 
-    val paths = inputPaths.map(_.getPath)
+    val rddInputPaths = inputPaths.map(_.getPath)
 
-    if (paths.nonEmpty) {
-      FileInputFormat.setInputPaths(job, paths: _*)
+    if (rddInputPaths.nonEmpty) {
```
Comment (chenghao-intel) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala]: A better place probably in `HadoopFsRelation.cachedLeafStatuses`? So all of the hadoopFsReleation will be benefit from the checking.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala [hunk 2]
```diff
@@ -83,10 +83,13 @@ private[sql] class JSONRelation(
...
+      FileInputFormat.setInputPaths(job, rddInputPaths: _*)
+    } else {
+      throw new RuntimeException("Input paths do not exist or are empty directories, "
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala]: Use `require` to write this test. In general `RuntimeException` is not something you should throw directly.

**Comment by srowen** [comment]
@zjffdu are you able to update this PR?

**Comment by JoshRosen** [comment]
It looks like this PR and #8899 are trying to solve the same issue. I'm going to link those JIRA tickets as duplicates.

**Comment by zjffdu** [comment]
Close this PR as it is a duplicate of #8899

**Comment by rxin** [comment]
@zjffdu can you reopen this?

**Comment by zjffdu** [comment]
It's corrupted with other commits after I rebase, close it and will create another PR

**Comment by l15k4** [comment]
Hey, does this support hadoop Globs and Wildcards? For which release this has been fixed? I guess 1.6, right?

**Comment by l15k4** [comment]
I just upgraded to to hadoop 2.6.4 and Spark 1.6 and I still get : 
```
java.io.IOException: No input paths specified in job
	at org.apache.hadoop.mapred.FileInputFormat.listStatus(FileInputFormat.java:201)
	at org.apache.hadoop.mapred.FileInputFormat.getSplits(FileInputFormat.java:313)
	at org.apache.spark.rdd.HadoopRDD.getPartitions(HadoopRDD.scala:199)
	at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:239)
```

I do : 
```
sqlContext.read.json("/mnt/views-p/base/2016/01/*/*-xyz.json").rdd
```

**Pull request closed without merging**